gate.lic

```diff
@@ -0,0 +1,194 @@
+        return false
+      end
+    end
+    return moon
+  end
+
+  def check_buffs
+    wait_for_script_to_complete('buff',['gate'])
+  end
+
+  def send_message(character, message)
+    @lnet.unique_buffer.push("chat to #{character} #{message}")
+  end
+
+  # copied from crossing-trianing
+  def check_discern(data)
+    UserVars.discerns = {} unless UserVars.discerns
+    discern_data = UserVars.discerns[data['abbrev']] || {}
+    if data['symbiosis']
+      if discern_data.empty? || discern_data['min'].nil?
+        /requires at minimum (\d+) mana streams/ =~ bput("discern #{data['abbrev']}", 'requires at minimum \d+ mana streams')
+        discern_data['mana'] = Regexp.last_match(1).to_i
+        discern_data['cambrinth'] = nil
+        discern_data['min'] = Regexp.last_match(1).to_i
+        discern_data['more'] = 0
+      end
+      calculate_mana(discern_data['min'], discern_data['more'], discern_data, false)
+    elsif discern_data.empty? || discern_data['time_stamp'].nil? || Time.now - discern_data['time_stamp'] > 24 * 60 * 60 || !discern_data['more'].nil?
+      discern_data['time_stamp'] = Time.now
+      case discern = bput("discern #{data['abbrev']}", 'The spell requires at minimum \d+ mana streams and you think you can reinforce it with \d+ more', 'You don\'t think you are able to cast this spell', 'You have no idea how to cast that spell')
+      when /you don't think you are able/i, 'You have no idea how to cast that spell'
+        discern_data['mana'] = 1
+        discern_data['cambrinth'] = nil
+      else
+        discern =~ /minimum (\d+) mana streams and you think you can reinforce it with (\d+) more/i
+        calculate_mana(Regexp.last_match(1).to_i, Regexp.last_match(2).to_i, discern_data, data['cyclic'])
+      end
+    end
+    pause 1
+    waitrt?
+    UserVars.discerns[data['abbrev']] = discern_data
+    data['mana'] = discern_data['mana']
+    data['cambrinth'] = discern_data['cambrinth']
+  end
+
+def calculate_mana(min, more, discern_data, cyclic)
+    total = min + more
+    total = (total * @settings.prep_scaling_factor).floor
+    discern_data['mana'] = [(total / 4.0).ceil, min].max
+    remaining = total - discern_data['mana']
+    if remaining > @settings.cambrinth_cap
+      discern_data['mana'] = discern_data['mana'] + (remaining - @settings.cambrinth_cap)
+      remaining = total - discern_data['mana']
+    end
+    if cyclic
+      discern_data['cambrinth'] = nil
+      discern_data['mana'] = discern_data['mana'] + remaining
+    elsif remaining > 0
+      discern_data['cambrinth'] = []
+      step_size = (remaining / 3.0).ceil
+      while remaining > 0
+        discern_data['cambrinth'] << [remaining, step_size].min
+        remaining -= step_size
+      end
+    else
+      discern_data['cambrinth'] = nil
+    end
+  end
+
+  def cast_spell(spell, mana, target, snap = false)
+
+    prepare?(spell, mana, false)
+    snap ? pause(5) : waitcastrt?
```

Comment:
Is this logic still necessary, or will the `prep_time` attribute on a spell definition serve the same purpose?

Reply:
I was trying to avoid yaml dependencies for this since this method is used for some explicit casting (i.e low prep CV, etc)